From 21e110f3d40ff0131fd3d91cdf7c3020aaa12305 Mon Sep 17 00:00:00 2001
From: Mauri de Souza Meneguzzo <mauri870@gmail.com>
Date: Tue, 28 Apr 2026 10:36:09 -0300
Subject: [PATCH] Enable posted MSI by default

---
 drivers/iommu/irq_remapping.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iommu/irq_remapping.c b/drivers/iommu/irq_remapping.c
index c2443659812a..0e71a648f403 100644
--- a/drivers/iommu/irq_remapping.c
+++ b/drivers/iommu/irq_remapping.c
@@ -24,7 +24,7 @@ int no_x2apic_optout;
 
 int disable_irq_post = 0;
 
-bool enable_posted_msi __ro_after_init;
+bool enable_posted_msi __ro_after_init = true;
 
 static int disable_irq_remap;
 static struct irq_remap_ops *remap_ops;
-- 
2.54.0

