From 5221af0bb72620525ba10e1effea9f3b6819d4a8 Mon Sep 17 00:00:00 2001
From: Mauri de Souza Meneguzzo <mauri870@gmail.com>
Date: Tue, 28 Apr 2026 10:37:50 -0300
Subject: [PATCH] Improve idle CPU load balancing

---
 kernel/sched/fair.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index e810385c9c96..7d80598c0c63 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -13017,7 +13017,7 @@ static int sched_balance_newidle(struct rq *this_rq, struct rq_flags *rf)
 
 		update_next_balance(sd, &next_balance);
 
-		if (this_rq->avg_idle < curr_cost + sd->max_newidle_lb_cost)
+		if (this_rq->avg_idle/2 < curr_cost + sd->max_newidle_lb_cost)
 			break;
 
 		if (sd->flags & SD_BALANCE_NEWIDLE) {
-- 
2.54.0

